### PR TITLE
database handlers support private-key pair as sslkey and sslcert

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
#### What does this PR do (include background context, if relevant)?

Adds support for sslkey and sslcert for mysql and postgres handlers.

**TODO:**  Mysql validates the sslcert and sslkey against the CA, Postgres does not. This needs to be  documented!

#### What ticket does this PR close?
Connected to #549 
#### What is the status of the manual tests?
Have you run the following manual tests to verify existing functionality continues to function as expected?

N/A

- [ ] Manually tested [K8s CRDs](https://github.com/cyberark/secretless-broker/tree/master/test/manual/k8s_crds)
- [ ] Manually tested [Keychain provider](https://github.com/cyberark/secretless-broker/tree/master/test/manual/keychain_provider)
- [ ] Manually run the [K8s demo](https://github.com/cyberark/secretless-broker/tree/master/demos/k8s-demo)
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local Secretless Broker image build of your branch
- [ ] Manually run the [full demo](https://github.com/cyberark/secretless-broker/tree/master/demos/full-demo) (optional)

If this feature does not have any/sufficent automated tests, have you created/updated a folder in `test/manual` that includes:
- [ ] An updated README with instructions on how to manually test this feature
- [ ] Utility `start` and `stop` scripts to spin up and tear down the test environments
- [ ] A `test` script to run some basic manual tests (optional; if does not exist, the README should have detailed instructions)

### CODE REVIEW

File: test/mysql_handler/start [hunk 1]
```diff
@@ -10,7 +10,11 @@ done
 
 ./stop
 
+# ./ssl is REQUIRED during the build phase to make test ssl cert artifacts available to the docker context
```
Comment (jonahx) [inline test/mysql_handler/start]: Let's add a note about how the certs are stored in util/ssl beause the same ones are shared, and we copy them here only while the test is running.

File: test/mysql_handler/tests/ssl_test.go [hunk 2]
```diff
@@ -141,6 +141,50 @@ func TestSSL(t *testing.T) {
 				SSLRootCertType: Undefined,
 			},
 		},
+		{
+			TestDefinition: TestDefinition{
+				Description:         "server_tls, sslmode=verify-ca, sslrootcert=valid, sslkey=malformed, sslcert=malformed",
```
Comment (jonahx) [inline test/mysql_handler/tests/ssl_test.go]: Per our conversation today, let's change this description to be more human friendly / rspec like:

> It should error when the certificate or key is malformed

Same comment applies to other descriptions too.

File: test/pg_handler/tests/ssl_test.go [hunk 3]
```diff
@@ -145,6 +145,49 @@ func TestSSL(t *testing.T) {
 				SSLRootCertType: Undefined,
 			},
 		},
+		{
+			TestDefinition: TestDefinition{
+				Description:         "server_tls, sslmode=verify-ca, sslrootcert=valid, sslkey=malformed, sslcert=malformed",
```
Comment (jonahx) [inline test/pg_handler/tests/ssl_test.go]: Same comment here as above re: human friendly descriptions

File: test/pg_handler/tests/ssl_test.go [hunk 4]
```diff
@@ -145,6 +145,49 @@ func TestSSL(t *testing.T) {
...
+				ListenerType:      TCP,
+				ServerTLSType:     TLS,
+				SSLModeType:       VerifyCA,
+				SSLRootCertType:   Valid,
+				SSLPrivateKeyType: PrivateKeyValid,
+				SSLPublicCertType: PublicCertValid,
+			},
+		},
+		{
+			TestDefinition: TestDefinition{
```
Comment (jonahx) [inline test/pg_handler/tests/ssl_test.go]: This is probably my own misunderstanding, but it's confusing that the cert here is _not_ signed by the CA, the sslModeType is `VerifyCA`, and it's expected to pass.  Can we add a note explaining this.

File: test/util/cfssl/ca-config.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (jonahx) [inline test/util/cfssl/ca-config.json]: Let's add a link to the relevant cfssl docs at the top here.

File: test/util/cfssl/ca-csr.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (jonahx) [inline test/util/cfssl/ca-csr.json]: Same here.

File: test/util/cfssl/generate_certificates.sh [hunk 7]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
```
Comment (jonahx) [inline test/util/cfssl/generate_certificates.sh]: this is looking good now.

File: test/util/ssl/ca-invalid.pem [hunk 8]
```diff
@@ -1,3 +1,6 @@
+// File generated by ROOT/util/cfssl/generate_certificates.sh
+// DO NOT EDIT
```
Comment (jonahx) [inline test/util/ssl/ca-invalid.pem]: 👍

File: test/util/test/config_generator.go [hunk 9]
```diff
@@ -105,59 +105,71 @@ func GenerateConfigurations() (config.Config, LiveConfigurations) {
...
+									SSLPrivateKeyType: sslPrivateKeyValue,
+									SSLPublicCertType: sslPublicCertValue,
+								},
+								ConnectionPort: connectionPort,
+							}
 
-					// sslRootCertTypeValue
-					handler.Credentials = append(handler.Credentials, sslRootCertTypeValue.toConfigVariable())
+							// sslRootCertTypeValue
+							handler.Credentials = append(handler.Credentials, sslRootCertTypeValue.toConfigVariable())
```
Comment (jonahx) [inline test/util/test/config_generator.go]: `append` takes multiple arguments -- any reason not to combine all these append statements into one?

File: test/util/cfssl/ca-config.json (hunk 5 above)
Comment (doodlesbykumbi) [reply test/util/cfssl/ca-config.json]: JSON doesn't support comments

File: test/pg_handler/tests/ssl_test.go (hunk 4 above)
Comment (doodlesbykumbi) [reply test/pg_handler/tests/ssl_test.go]: VerifyCA is to do with `sslmode`.
It means the client wants to validate the certificate of the server against a certificate the client will provide.

This is distinct from the validation of client private key and certificate. Notably, the client can not configure this. It's at the server's discretion. E.g. mysql checks to ensure the client certificate has been signed by the root CA on the server.

I personally don't think here is the place to add this information.

File: test/util/test/config_generator.go (hunk 9 above)
Comment (doodlesbykumbi) [reply test/util/test/config_generator.go]: that's true. changed

File: test/mysql_handler/tests/ssl_test.go (hunk 2 above)
Comment (doodlesbykumbi) [reply test/mysql_handler/tests/ssl_test.go]: actually all the tests would need to be changed to this format. i'd opt for doing this in a follow up PR

File: bin/reflex [hunk 10]
```diff
@@ -3,4 +3,4 @@ set -e
 
 # watch for changes to source files then rebuild and run secretless-broker
 reflex -r '\.go$' -R '\_test.go$' -s -- \
-  bash -c "rm -rf /sock/*; go run cmd/secretless-broker/main.go -f /secretless.yml --watch"
+  bash -c "rm -rf /sock/*; rm -rf /sock/.s*; go run cmd/secretless-broker/main.go -f /secretless.yml --watch"
```
[the remark this answers is not in the document]
Comment (doodlesbykumbi) [reply bin/reflex]: sorted

**Pull request merged**